pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -412,6 +412,14 @@ func (os *OpenStack) DiskIsAttached(instanceID, volumeID string) (bool, error) {
 		return false, err
 	}
 
+	if instanceID == "" {
+		if volume.AttachedServerId != "" {
+			return true, nil
```

Comment:
Is that confused? Is it need to add another func to check whether disk is attached?